examples/resources/coder_app/resource.tf

```diff
@@ -11,11 +11,14 @@ EOF
 }
 
 resource "coder_app" "code-server" {
-  agent_id          = coder_agent.dev.id
-  name              = "VS Code"
-  icon              = data.coder_workspace.me.access_url + "/icons/vscode.svg"
-  url               = "http://localhost:13337"
-  relative_path     = true
+  agent_id      = coder_agent.dev.id
+  name          = "VS Code"
+  icon          = data.coder_workspace.me.access_url + "/icons/vscode.svg"
+  url           = "http://localhost:13337"
+  relative_path = true
+  healthcheck_url = "http://localhost:13337/healthz"
+  healthcheck_interval = 5
+  healthcheck_threshold = 6
```

Comment:
Should we put these inside a `healthcheck` block instead?
```
healthcheck {
  url = "http://..." 
  interval = 5
  threshold = 6
}
```

Reply:
I considered this but ultimately didn't because I didn't want to make a new codersdk type for it on the api side and wanted the API and the terraform to share the same schema if possible. If we are cool with the TF being different from the API type schema I'm down for just the TF side.

Reply:
Why not create a new type in `codersdk` for it? It seems kinda like a separate entity, and it makes more sense from my POV for this to be a block.

Reply:
It also cleanly separates that health checking is disabled if the block isn't specified, which is kinda nice.

Reply:
Sure I'm down if we like that

Reply:
I'm personally a fan, and I believe @deansheather is too based on the :+1: :eyes: 

shiiip it